### PR TITLE
add notes about transitive deps

### PR DESCRIPTION
**Pull request by pongad**

### CODE REVIEW

File: library-best-practices/JLBP-1.md [hunk 1]
```diff
@@ -2,13 +2,19 @@
...
+Every dependency of a library is a liability of both
+that library and its consumers.
+
+Adding a dependency for a large amount of functionality or
+for a complex functionality may be OK,
+but avoid pulling in dependencies just to save a few lines of code.
 
 Some specific notes about minimizing dependencies:
 
+- Remember that liability, like dependencies, is transitive.
```
Comment (elharo) [inline library-best-practices/JLBP-1.md]: I'm not sure I follow what you're trying to say here.

File: library-best-practices/JLBP-1.md [hunk 2]
```diff
@@ -2,13 +2,19 @@
...
 Use the minimum number of dependencies that is reasonable.
-Adding a dependency for a large amount of functionality may be ok,
-but avoid pulling in dependencies just to save a few lines of code,
-because every dependency of a library is a liability of both
-that library and that library's consumers.
+Every dependency of a library is a liability of both
+that library and its consumers.
+
+Adding a dependency for a large amount of functionality or
+for a complex functionality may be OK,
```
Comment (elharo) [inline library-best-practices/JLBP-1.md]: "for a complex functionality" is a little awkward. Maybe "for something that is difficult and complicated to do"

File: library-best-practices/JLBP-1.md [hunk 3]
```diff
@@ -2,13 +2,19 @@
...
+but avoid pulling in dependencies just to save a few lines of code.
```
Comment (elharo) [inline library-best-practices/JLBP-1.md]: perhaps "avoid adding a dependency just to save a few lines of code"

File: library-best-practices/JLBP-1.md (hunk 1 above)
Comment (pongad) [reply library-best-practices/JLBP-1.md]: Would "remember you're also paying the cost of transitive dependencies" be better?

Comment (pongad) [reply library-best-practices/JLBP-1.md]: done

File: library-best-practices/JLBP-1.md (hunk 2 above)
Comment (pongad) [reply library-best-practices/JLBP-1.md]: done

File: library-best-practices/JLBP-1.md (hunk 3 above)
Comment (pongad) [reply library-best-practices/JLBP-1.md]: done

**Comment by pongad** [comment]
Thank you for review @elharo PTAL

**Comment by pongad** [comment]
My pleasure! If this looks good to you, I think you need to merge yourself. I don't have write access to this repo :P

**Pull request merged**